### ISSUE TITLE
Detecting unknown objects

### ISSUE THREAD

**Issue by hamdii02** [AlexeyAB/darknet#7685]
I am trying to find a solution to detect the classes of the COCO dataset and detect any new objects in the investigated area. In other words, the model should detect any new objects and states that this is an unknown object.
Is this possible by using YOLO or any other techniques ?? (I know that from a deep learning point of vue the model will detect only the objects that he knows during training but am asking about any new trick to perform this task)
Thank you

**Comment by ekesdf** [comment]
No, on the current point of machine learning you always need to train the model for the objects you want to train or even similar object. I got  around this by just detecting any object in the image and I then build and trained my own custom classification model, so I can detect even some "unknown" objects and then classify them later. So I could detect the thing my classification model don't know, but my detection model can detect. 

So you can "train" your model to detect some "unknown" model and classify them later for that I would use Segmentation to separate the Object from the Background. 

Maybe you can just upload some images so I can check what object detection strategy fits your purpose best :)

**Comment by hamdii02** [comment]
Thank you for responding
In my case I have several object inside a room and I want to detect all the objects before proceeding with the object recognition. My model is not trained to recognize all the objects inside the room, so I just want to detect all the object then I will run my model, and hence I can discover the unknown objects (objects who are not recognized by my model)
Thank you

**Comment by ekesdf** [comment]
Is the "background" all the same color as a white wall or is there noise in the background ?
**Like in this Image ** 
![image](https://user-images.githubusercontent.com/64316659/117309139-72cdbf00-ae82-11eb-8269-8910ca7ebd87.png)

Yea just train your object detection model with all objects in one class that should help to detect unseen objects, so your model basically learn to decide between a random object and the background.

**Comment by hamdii02** [comment]
No, the camera is a CCTV camera in a room or a kitchen, etc., therefore the background has noise and it is not homogeneous.
This is the problem that I am facing here. Moreover, even if I have a background with no noise, and I worked on finding a solution using image processing without including any AI algorithms, I will for sure suffer from the problem of luminosity because the area could be a living room or a kitchen with windows.
example of use case: 
![depositphotos_212101224-stock-photo-interior-living-room-view-cctv](https://user-images.githubusercontent.com/46610259/117311789-6f373980-ae7c-11eb-9b70-6ea0b47b0f90.jpg)

**Comment by haviduck** [comment]
> I am trying to find a solution to detect the classes of the COCO dataset and detect any new objects in the investigated area. In other words, the model should detect any new objects and states that this is an unknown object.
> Is this possible by using YOLO or any other techniques ?? (I know that from a deep learning point of vue the model will detect only the objects that he knows during training but am asking about any new trick to perform this task)
> Thank you

opencvs absdiff function lets you match a baseline frame with a new frame. its used like that for motion detection atleast.
http://www.robindavid.fr/opencv-tutorial/chapter10-movement-detection-with-background.html
